src/Core/Resolvers/MultipleCreateOrderHelper.cs

```diff
@@ -0,0 +1,340 @@
+
+                // Update whether source/target contains any relationship field which is autogenerated.
+                doesSourceContainAnyAutogenRelationshipField = doesSourceContainAnyAutogenRelationshipField || isSourceRelationshipColumnAutogenerated;
+                doesTargetContainAnyAutogenRelationshipField = doesTargetContainAnyAutogenRelationshipField || isTargetRelationshipColumnAutogenerated;
+
+                // When both source/target entities contain an autogenerated relationship field, we cannot choose one entity
+                // as the referencing entity. This is because for a referencing entity, the values for all the referencing fields
+                // should be derived from the insertion in the referenced entity. However, here we would not be able to assign
+                // value to an autogenerated relationship field in the referencing entity.
+                if (doesSourceContainAnyAutogenRelationshipField && doesTargetContainAnyAutogenRelationshipField)
+                {
+                    throw new DataApiBuilderException(
+                        message: $"Both source entity: {sourceEntityName} and target entity: {targetEntityName} contain autogenerated fields for relationship: {relationshipName} at level: {nestingLevel}",
+                        statusCode: HttpStatusCode.BadRequest,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                }
+
+                // If the source entity contains a relationship field in request body which suggests us to do the insertion first in source entity,
+                // and there is an autogenerated relationship field in the target entity which suggests us to do insertion first in target entity,
+                // we cannot determine a valid order of insertion.
+                if (doesSourceBodyContainAnyRelationshipField && doesTargetContainAnyAutogenRelationshipField)
+                {
+                    throw new DataApiBuilderException(
+                        message: $"Input for source entity: {sourceEntityName} for the relationship: {relationshipName} at level: {nestingLevel} cannot contain the field: {relationshipFieldInSource}.",
+                        statusCode: HttpStatusCode.BadRequest,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                }
+
+                // If the target entity contains a relationship field in request body which suggests us to do the insertion first in target entity,
+                // and there is an autogenerated relationship field in the source entity which suggests us to do insertion first in source entity,
+                // we cannot determine a valid order of insertion.
+                if (doesTargetBodyContainAnyRelationshipField && doesSourceContainAnyAutogenRelationshipField)
```

Comment:
I am not sure if `we insert the target entity first` conveys the right message. We are inserting a record in the entity rather than inserting an entity.